packages/core/src/helpers/createDinero.ts

```diff
@@ -23,7 +23,11 @@ export function createDinero<TAmount>({
       toJSON() {
         return {
           amount,
-          currency,
+          currency: {
+            code: currency.code,
+            base: currency.base,
+            exponent: currency.exponent,
+          },
```

Comment:
This works fine for `toJSON` but `onCreate` in l. 18 still receives the full currency object.

My recommendation would be to destructure each property of `currency` in the parameters, then create a `currency` constant in the function body from these destructured properties. This way the rest of the code doesn't have to change.

Performance-wise, this should be negligible.

Reply:
Thanks and thanks for this work!